src/desktop/lib/build_server_app_context.ts

```diff
@@ -0,0 +1,22 @@
+// @ts-ignore
+import mediator from "desktop/lib/mediator.coffee"
+import { Request as ExpressRequest, Response as ExpressResponse } from "express"
+
+interface Request extends ExpressRequest {
+  user: any
+}
+type Response = ExpressResponse
+
+/**
+ * Builds initial context for Reaction components from server load. Put commonly
+ * required context here.
+ *
+ * @param req Request passed to Express callback.
+ * @param res Response passed to Express callback.
+ */
+export const buildServerAppContext = (req: Request, res: Response) => ({
```

Comment:
Thinking out loud here... would this make sense to live in reaction? `buildServerApp`already lives there. This is an extension of that by providing default context via grabbing some info off of `req` and `res`.. 

What if we just passed `req` and `res` directly to build server app? It could build up the default context and we could spread out any specified context values to that. It'd also give the ability to do other express-y things later if needed...

I'm not convienced that the above suggestion is the best thing to do. This current approach works pretty well. Still, interested in thoughts. @ashfurrow, @damassi?

Reply:
Hmm, I mean, you both have more experience with this than I do, but it currently feels like Reaction and Force have a clear boundary between who does the Express request-level stuff and who stays in the React-level abstractions. Moving this code to within Reaction would break that existing division. Again, from my perspective.

Reply:
Agree, I don't think that this should go in Reaction since Force handles the req / res flow. (This will be changing in the future though as we agreed in the last FE working group that new apps will live in a shell on the reaction side in the future, but this still needs to be built.)

@ashfurrow - One thing however just to clarify the use of `context`. In 

```js
buildServerApp({
  context: {
    // anything here gets injected into React's context 
    // and is available globally via <ContextConsumer> in reaction
  }
})
```

Different apps may have different needs, so in the future this might be unwound. `context` just a bus for passing things across the force / reaction divide, or if building an app in reaction (since `context` here also applies to `buildClientApp`), a means for injecting global values.

Reply:
Also, I must note you're becoming quite the web engineer :) Thanks for giving this a good 👁